### PR TITLE
Add localhost into hosts based on network mode

### LINKED ISSUE #11411: There is no localhost on /etc/hosts
/kind bug

I have setup rootless podman and after running a container everything works find but in the `/etc/hosts/` I have the following content:

```
# Static table lookup for hostnames.
# See hosts(5) for details.
# used by slirp4netns
10.0.2.100      b932ec01d49e compassionate_visvesvaraya
10.0.2.2 host.containers.internal
```

but I think it should be:

```
# Static table lookup for hostnames.
# See hosts(5) for details.
127.0.0.1 localhost
# used by slirp4netns
10.0.2.100      b932ec01d49e compassionate_visvesvaraya
10.0.2.2 host.containers.internal
```

and because of that there is no resolution from localhost to 127.0.0.1.

### PR DESCRIPTION
**Pull request by 1995parham**
fixes #11411

### CODE REVIEW

**Comment by 1995parham** [comment]
I cannot find any existing test for `container_linux_internal` and `/etc/hosts`. I think if you agree I can add a test to create a temporary file to works as `hosts` and then check the `generateHosts` function but this function covers many different logics.

File: libpod/container_internal_linux.go [hunk 1]
```diff
@@ -1942,9 +1942,24 @@ func (c *Container) generateHosts(path string) (string, error) {
...
+
 	return c.writeStringToRundir("hosts", hosts)
 }
 
+// based on networking mode we may want to append the localhost
+// if there isn't any record for it and also this shoud happen
+// in slirp4netns and similar network modes.
+func (c *Container) appendLocalhost(hosts string) string {
+	if !strings.Contains(hosts, "localhost") &&
+		c.config.NetMode.IsSlirp4netns() {
```
Comment (Luap99) [inline libpod/container_internal_linux.go:1956]: I think `!c.config.NetMode.IsHost()` is better. Only for host we don't want to add this.

Comment (1995parham) [reply libpod/container_internal_linux.go]: Thanks for mentioning this, I have update code and tests.

**Comment by rhatdan** [comment]
LGTM

**Comment by Luap99** [comment]
Can you squash your commits into one please.

**Comment by 1995parham** [comment]
@Luap99 sure I will squash them now.

**Pull request merged**